htmltools/_core.py

```diff
@@ -87,7 +97,7 @@ class MetadataNode:
 
 TagT = TypeVar("TagT", bound="Tag")
 
-TagAttrValue = Union[str, float, bool, None]
+TagAttrValue = Union[str, float, bool, "HTML", None]
```

Comment:
I do not believe we should make a `is_tag_attr_value()` method as `consolidate_attrs()` should internally handle it. 

For now, we will skip making this helper method.